lib/time.dart

```diff
@@ -0,0 +1,52 @@
+library w_common.timestamp;
+
+import 'package:intl/intl.dart';
+
+/// The format of a timestamp with no date.
+DateFormat timeFormat = new DateFormat('h:mma');
+
+/// The format of a weekday with no time of day.
+DateFormat weekdayFormat = new DateFormat.EEEE();
+
+/// The format of a month and day with no time of day.
+DateFormat monthDayFormat = new DateFormat.MMMMd();
+
+/// The format of the full date with no time of day.
+DateFormat yearMonthDayFormat = new DateFormat.yMMMd();
+
+/// formating of string in our db
+DateFormat dbFormat = new DateFormat("y-M-dd HH:mm:ss");
+
+/// Formats a DateTime into the 'X ago' string format.
+String formatTimeDifference(DateTime time, {DateTime now}) {
+  now ??= new DateTime.now();
+  final timeOfDay = timeFormat.format(time).toLowerCase();
+  final deltaDays = now.difference(time).inDays.abs();
+
+  if (deltaDays < 1 && now.day == time.day) {
+    // "Today, XX:XXam"
+    return 'Today, $timeOfDay';
+  }
+
+  if (deltaDays < 2 && now.weekday == (time.weekday + 1) % 7) {
+    // "Yesterday, XX:XXam"
+    return 'Yesterday, $timeOfDay';
+  }
+
+  // Weekday check prevents abiguity between comments
```

Comment:
ambiguity* I know you just moved this over and inherited this spelling mistake, but might as well fix it now :)

Reply:
Yep, that seems like a good idea